packages/flutter/lib/src/widgets/pinned_header_sliver.dart

```diff
@@ -0,0 +1,74 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+
+import 'dart:math' as math;
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/rendering.dart';
+
+import 'framework.dart';
+
+/// A sliver that keeps its Widget child at the top of the a [CustomScrollView].
+///
+/// This sliver is preferable to the general purpose [SliverPersistentHeader]
```

Comment:
Should this talk more about when to use one or the other?

Should folks expect this one to expand/stretch/collapse? Or have a shadow when scrolled under?

I wonder if users will request this sliver to ultimately adopt all the existing features of the pinned persistent header we already have. 

I follow that this one can adapt to the size of the child, where the existing one does not and takes the size from the delegate. for posterity, why add a new sliver instead of expanding on the existing one?

Reply:
Just sharing more thoughts.. will folks expect us to add a FloatingHeaderSliver to use instead of a floating SliverPersistentHeader? FloatingPinnerHeaderSliver?

Reply:
I think it's better not to bundle all the possible behaviors into one sliver. Having one sliver per behavior keeps the APIs simple and scales nicely when new behaviors are added (in their own classes).

I did write a FloatingHeaderSliver but forgot to include it in this raft of slivers. I will add it.